src/docfx/build/metadata/ContributionInfo.cs

```diff
@@ -190,15 +197,38 @@ private GitUserInfo ToGitUserInfo(UserProfile profile)
             {
                 var pathToDocset = group.Select(pair => pair.file.FilePath).ToList();
                 var pathToRepo = group.Select(pair => pair.pathToRepo).ToList();
-                var repoPath = group.Key.RepositoryPath;
-                var commitsList = GitUtility.GetCommits(repoPath, pathToRepo);
+                var repo = group.Key;
+                var repoPath = repo.RepositoryPath;
+                var (commitsByFile, allCommits) = GitUtility.GetCommits(repoPath, pathToRepo);
                 for (var i = 0; i < pathToDocset.Count; i++)
                 {
-                    result.Add(pathToDocset[i], commitsList[i]);
+                    result.Add(pathToDocset[i], commitsByFile[i]);
+                }
+                foreach (var commit in allCommits)
+                {
+                    try
+                    {
+                        await UpdateCache(commit, repo);
```

Comment:
>UpdateCache [](start = 30, length = 11)

put these logic in UserProfileCache, contribution info should just read from the cache, UserProfileCache handles profile update internally.